### PR TITLE
Use upper case and improve focusing algorithm

### PR DESCRIPTION
**Pull request by Aaron2000119**
Closes https://github.com/Gamify-IT/issues/issues/456.
Closes https://github.com/Gamify-IT/issues/issues/502.

The focus algorithm should work now and every letter should be upper-case.

### CODE REVIEW

**Review by madebyTimo — APPROVED** [review]
Works perfect, code looks fine. You could outsource the methods so that the file is not so overloaded

File: src/components/InputField.vue [hunk 1]
```diff
@@ -44,9 +45,34 @@ watch(
...
+}
+
 async function keyDown(key: KeyboardEvent) {
   if (key.key.length == 1) {
-    crosswordTile.value.currentLetter = key.key;
+    crosswordTile.value.currentLetter = key.key.toUpperCase();
+
     let elementRight = document.getElementById(
       "inputField:x" +
         (crosswordTile.value.positionX + 1) +
```
Comment (delvh) [inline src/components/InputField.vue:78]:
```js
`inputField:x${crosswordTile.value.positionX + 1},y${crosswordTile.value.positionY}`
```
?

File: src/components/InputField.vue [hunk 2]
```diff
@@ -59,48 +85,44 @@ async function keyDown(key: KeyboardEvent) {
         ",y" +
         (crosswordTile.value.positionY + 1)
```
Comment (delvh) [inline src/components/InputField.vue:86]: See above

File: src/components/InputField.vue [hunk 3]
```diff
@@ -117,9 +139,153 @@ async function keyDown(key: KeyboardEvent) {
...
+      case "ArrowUp": {
+        emit("direction", "down");
+        pressUpArrow();
+        break;
+      }
+      case "Tab": {
+        pressTab();
+        break;
+      }
+    }
```
Comment (delvh) [inline src/components/InputField.vue:168]: What about extracting this switch into a separate method `handleButtonPress` or similar?

File: src/components/InputField.vue [hunk 4]
```diff
@@ -117,9 +139,153 @@ async function keyDown(key: KeyboardEvent) {
...
+  }
+
+  function pressTab() {
+    let elementRight = document.getElementById(
```
Comment (delvh) [inline src/components/InputField.vue:172]: At this point, I recommend a helper method that gives you a tile given its x and y coordinates.

File: src/components/InputField.vue [hunk 5]
```diff
@@ -44,63 +45,80 @@ watch(
   { deep: true }
 );
 
+function isLeftStartPoint(): boolean {
+  let leftStartPoint = false;
+  if (
+    crossword.value[crosswordTile.value.positionY][
+      crosswordTile.value.positionX - 1
+    ].startPoint
```
Comment (delvh) [inline src/components/InputField.vue:53]:
```suggestion
    getCrosswordElement(crosswordTile.value.positionX - 1, crosswordTile.value.positionY).startPoint
```

Comment (Aaron2000119) [reply src/components/InputField.vue:53]: getCrosswordElement and crossword.value[][] are different elements. getCrosswordElement gets the specific Tile on the dom and crossword.value[][] gets the object in the crossword array.

**Review by delvh — APPROVED** [review]

**Pull request merged**